pkg/providers/v1/aws.go

```diff
@@ -3508,6 +3523,40 @@ func (c *Cloud) findSubnets() ([]*ec2.Subnet, error) {
 	return subnets, nil
 }
 
+// Returns a mapping between availability zone names and their types
+// Zone will not be included in the map in case it was not found in AWS by name
+func (c *Cloud) getZoneTypesByName(azNames []string) (map[string]string, error) {
+	if len(azNames) == 0 {
+		// if az names slice is empty, no need to make a request, return early with empty map
+		return map[string]string{}, nil
+	}
+	azFilter := newEc2Filter("zone-name", azNames...)
+	azRequest := &ec2.DescribeAvailabilityZonesInput{}
+	azRequest.Filters = []*ec2.Filter{azFilter}
+
+	azs, err := c.ec2.DescribeAvailabilityZones(azRequest)
+	if err != nil {
+		if awsErr, ok := err.(awserr.RequestFailure); ok && awsErr.Code() == "UnauthorizedOperation" {
```

Comment:
If you wrap the inner error then you could do something like this to achieve the same
```
awsErr := awserr.RequestFailure{}
if errors.As(err, awsErr) && awsErr.Code() ...
```

Reply:
I looked over the file, and the only pattern to check aws errors is through type assertions. I would prefer to keep it consistent and not introduce a new pattern here. 

But i agree, wrap errors with "%w" and use `errors.As` will be better, but i suggest to address this in other PR and make this same across errors which is returning from this `ec2` sdk wrapper.

Reply:
If this keeps it consistent then fair enough